### ISSUE TITLE
crossentropy should have label smoothing 

### ISSUE THREAD

**Issue by sambitdash** [FluxML/Flux.jl#1016]
`crossentropy` method should support label smoothing. The equivalent API can be seen in TensorFlow at:

https://www.tensorflow.org/api_docs/python/tf/keras/losses/BinaryCrossentropy

**Comment by CarloLucibello** [comment]
Do we need this? It is not very common AFAIK, pytorch doesn't implement it, and users can implement it very easily by passing to the loss function the shifted labels

**Comment by sambitdash** [comment]
> Do we need this? It is not very common AFAIK, pytorch doesn't implement it, and users can implement it very easily by passing to the loss function the shifted labels

Enhancement request is getting implemented in pytorch. 

https://github.com/pytorch/pytorch/issues/7455
